### PR TITLE
Move CombineValues override to non portable overrides

### PR DESCRIPTION
**Pull request by damccorm**
Right now, running the following pipeline on Dataflow:

```
import logging

import apache_beam as beam
from apache_beam.options.pipeline_options import DebugOptions

def merge(vals):
    out = ""
    for v in vals:
        out += v
    return out

def run():
    pipeline_options = DebugOptions() 
    pipeline_options.dataflow_job_file = "path/to/dump/proto"

    with beam.Pipeline(options=pipeline_options) as p:
        (p
            | beam.Create([("key1", "foo"), ("key2", "bar"), ("key1", "baz")])
            | beam.GroupByKey()
            | beam.CombineValues(merge)
        )


if __name__ == '__main__':
    logging.getLogger().setLevel(logging.INFO)
    run()
```

succeeds when `use_runner_v2` is not included as an experiment, but fails when it is. This is because the translation being done generates an invalid graph for the `use_runner_v2` case: a portable pipeline shouldn't be using the overridden combine here.

This fixes the problem by moving the override to only occur when `use_runner_v2` is not included

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damccorm** [comment]
Run Python 3.9 PostCommit

**Comment by damccorm** [comment]
R: @lukecwik

**Review by lukecwik — CHANGES_REQUESTED** [review]
Can we add the described scenario as a new ValidatesRunner test to prevent a regression in the future?

**Comment by damccorm** [comment]
Run Python 3.8 PostCommit

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

**Comment by damccorm** [comment]
Run Python 3.8 PostCommit

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python 3.8 PostCommit

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

**Comment by damccorm** [comment]
Run Python 3.8 PostCommit

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

**Comment by AnandInguva** [comment]
stop reviewer notifications

**Comment by damccorm** [comment]
Run Python 3.8 PostCommit

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

File: sdks/python/apache_beam/examples/snippets/transforms/aggregation/combinevalues_it_test.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import pytest
+import unittest
+
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+from . import combinevalues
+
+class CombineValuesIT(unittest.TestCase):
```
Comment (lukecwik) [inline sdks/python/apache_beam/examples/snippets/transforms/aggregation/combinevalues_it_test.py:32]: I would suggest making this a validates runner test like:
https://github.com/apache/beam/blob/897fffdeda59cc2f121dd502e569770a897fc4ac/sdks/python/apache_beam/transforms/ptransform_test.py#L702

in ptransform_test.py

**Review by lukecwik — CHANGES_REQUESTED** [review]

**Comment by damccorm** [comment]
Run Python Dataflow V2 ValidatesRunner

**Comment by damccorm** [comment]
Run Python Dataflow ValidatesRunner

File: sdks/python/apache_beam/transforms/combiners_test.py [hunk 2]
```diff
@@ -873,6 +873,24 @@ def test_with_input_types_decorator_violation(self):
         _ = pc | beam.CombineGlobally(self.fn)
 
 
+@pytest.mark.it_validatesrunner
+class CombineValuesTest(unittest.TestCase):
+  def test_combinevalues(self):
```
Comment (lukecwik) [inline sdks/python/apache_beam/transforms/combiners_test.py:878]: nit:
```suggestion
  def test_gbk_immediately_followed_by_combine(self):
```

**Review by lukecwik — APPROVED** [review]
LGTM

**Pull request merged**